content/client-lib-development-guide/features.textile

```diff
@@ -515,7 +516,7 @@ h3(#realtime-presence). Presence
 ** @(RTP5c)@ In addition, when a channel becomes @ATTACHED@, the following must happen in regards to @Presence@:
 *** @(RTP5c1)@ If the "@resumed@ flag":#TH4 is true and no @SYNC@ is initiated as part of the attach, do nothing as there is no loss of continuity on the channel and no change to presence. The @PresenceMap@ is not affected and no members need to be re-entered
 *** @(RTP5c2)@ If a @SYNC@ is initiated as part of the attach, then once the @SYNC@ is complete, all members not present in the @PresenceMap@ but present in the "internal @PresenceMap@":#RTP17 must be re-entered automatically by the client using the @clientId@ and @data@ attributes from each. The members re-entered automatically must be removed from the "internal @PresenceMap@":#RTP17 ensuring that members present on the channel are constructed from presence events sent from Ably since the channel became @ATTACHED@
-*** @(RTP5c3)@ If any of the automatic @ENTER@ presence messages published in "RTP5c2":#RTP5c2 or "RTP5c3":#RTP5c3 fail, then the error message received from Ably should be emitted as an @ERROR@ event on the @Channel@ with Ably error code @91201@
+*** @(RTP5c3)@ If any of the automatic @ENTER@ presence messages published in "RTP5c2":#RTP5c2 fail, then an @UPDATE@ event should be emitted on the channel, with a @reason@ set to the error message received from Ably, wrapped in an @ErrorInfo@ with error code @91201@
```

Comment:
Re the error code, I'm now thinking it's probably be a mistake to wrap the error code from realtime. If someone wants to programatically do something useful with a failed re-enter, they need to know the actual reason it failed (which'll probably be one of the 9100x reasons), wrapping that in a general 'an auto-re-enter failed' code is probably unhelpful.

(of course, can still wrap just the message, `"An automatic re-enter attempt failed: #{error.message}"` etc, to make the log message more explicit)

Unless anyone objects, I'll just remove that bit